src/index.ts

```diff
@@ -53,10 +53,11 @@ const sidecarPlugin: JupyterFrontEndPlugin<void> = {
             });
           } else {
             const anchor = this.model.get('anchor') || 'right';
+            const ref = this.model.get('ref') || null;
```

Comment:
Now you'll need to wait for the ref to actually have been created. You need to make the render function `async`

```suggestion
            const ref = this.model.get('ref') || null;
            if (ref) {
              await ref.created();
            }
```